addons/account/models/account_move.py

```diff
@@ -3487,7 +3487,7 @@ def field_will_change(line, field_name):
             # As we can't express such synchronization as computed fields without cycling, we need to do it both
             # in onchange and in create/write. So, if something changed in accounting [resp. business] fields,
             # business [resp. accounting] fields are recomputed.
-            if any(field in cleaned_vals for field in ACCOUNTING_FIELDS):
+            if any(field in cleaned_vals or field in vals for field in ACCOUNTING_FIELDS):
```

Comment:
```suggestion
            if any(field in vals for field in ACCOUNTING_FIELDS):
```
then because `cleaned_vals` is a subset of `vals`